### PR TITLE
IGNITE-21317 Fix ItRebalanceDistributedTest

### PR DESCRIPTION
**Pull request by sanpwc**
https://issues.apache.org/jira/browse/IGNITE-21317

Table creation/Zone altering awaiting logic added.

### CODE REVIEW

File: modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java [hunk 1]
```diff
@@ -784,20 +780,23 @@ private void waitPartitionAssignmentsSyncedToExpected(int partNum, int replicasN
...
-        }
+        assertTrue(waitForCondition(
+                () -> {
+                    try {
+                        return nodes.stream().allMatch(n ->
+                                n.tableManager
+                                        .latestTables()
+                                        .get(getTableId(n, TABLE_NAME))
+                                        .internalTable()
+                                        .partitionRaftGroupService(0) != null
```
Comment (kgusakov) [inline modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java:791]: Why 0 and not partNum?

File: modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java [hunk 2]
```diff
@@ -519,8 +516,6 @@ void testDestroyPartitionStoragesOnRestartEvictedNode(TestInfo testInfo) throws
 
         createTableWithOnePartition(node, TABLE_NAME, ZONE_NAME, 3, true);
 
-        waitPartitionAssignmentsSyncedToExpected(0, 3);
```
Comment (kgusakov) [inline modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java:522]: Why did you remove it?

**Review by kgusakov — CHANGES_REQUESTED** [review]

File: modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java (hunk 2 above)
Comment (sanpwc) [reply modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java:522]: because createTableWithOnePartition includes waitPartitionAssignmentsSyncedToExpected
```
    private void createTableWithOnePartition(
            Node node,
            String tableName,
            String zoneName,
            int replicas,
            boolean testDataStorage
    ) throws Exception {
        createZone(node, zoneName, 1, replicas, testDataStorage);

        createTable(node, zoneName, tableName);

        waitPartitionAssignmentsSyncedToExpected(0, replicas);

        checkPartitionNodes(0, replicas);
    }
```

**Review by kgusakov — APPROVED** [review]
